core/src/test/scala/org/apache/spark/SparkConfSuite.scala

```diff
@@ -322,6 +324,291 @@ class SparkConfSuite extends SparkFunSuite with LocalSparkContext with ResetSyst
+  }
+
+  test("verify shuffle file buffer size configuration") {
+    val conf = new SparkConf(false)
+
+    // 2097151: Int.MaxValue / 1024
+    val maxvalue = 2097151L
+
+    Seq(
+      SHUFFLE_FILE_BUFFER_SIZE,
+      SHUFFLE_UNSAFE_FILE_OUTPUT_BUFFER_SIZE
+    ).foreach { config =>
+      assert(!conf.contains(config))
+
+      // min values
+      conf.set(config.key, "1k")
+      assert(conf.get(config) === 1)
+
+      // max values
+      conf.set(config.key, s"${maxvalue.toString}k")
+      assert(conf.get(config) === maxvalue)
+
+      // Verify exception values
+      Seq(
+        "0k",
+        "-1k",
+        s"${(maxvalue + 1).toString}k"
+      ).foreach { value =>
+        conf.set(config.key, value)
+        var sc0 = new SparkContext("local", "test", conf)
+        val a = sc0.parallelize(1 to 5, 201)
+        val b = a.map(x => (x, x*2))
+        val c = new ShuffledRDD[Int, Int, Int](b, new HashPartitioner(201))
+          .setSerializer(new KryoSerializer(conf))
+
+        val excMsg = intercept[SparkException] {
+          c.collect()
+        }.getMessage
+        // Caused by: java.lang.IllegalArgumentException:
+        // The file buffer size must be greater than 0 and less than ${Int.MaxValue / 1024}.
+        // or The buffer size must be greater than 0 and less than ${Int.MaxValue / 1024}.
+        assert(excMsg.contains("buffer size must be greater than 0 " +
+          s"and less than ${Int.MaxValue / 1024}."))
+        sc0.stop()
+        conf.remove(config)
+      }
+    }
+  }
+
+  test("verify spark.shuffle.spill.diskWriteBufferSize configuration") {
+    val conf = new SparkConf(false)
+
+    // 2147483647: Int.MaxValue
+    val maxvalue = 2147483647L
+
+    assert(!conf.contains(SHUFFLE_DISK_WRITE_BUFFER_SIZE))
+
+    // min values
+    conf.set(SHUFFLE_DISK_WRITE_BUFFER_SIZE.key, "1024B")
+    assert(conf.get(SHUFFLE_DISK_WRITE_BUFFER_SIZE) === 1024)
+
+    // max values
+    conf.set(SHUFFLE_DISK_WRITE_BUFFER_SIZE.key, s"${maxvalue.toString}B")
+    assert(conf.get(SHUFFLE_DISK_WRITE_BUFFER_SIZE) === maxvalue)
+
+    // Verify exception values
+    Seq(
+      "0B",
+      "-1B",
+      s"${(maxvalue + 1).toString}B"
+    ).foreach { value =>
+      conf.set(SHUFFLE_DISK_WRITE_BUFFER_SIZE.key, value)
+      var sc0 = new SparkContext("local", "test", conf)
+      val a = sc0.parallelize(1 to 5, 201)
+      val b = a.map(x => (x, x*2))
+      val c = new ShuffledRDD[Int, Int, Int](b, new HashPartitioner(201))
+        .setSerializer(new KryoSerializer(conf))
+
+      val excMsg = intercept[SparkException] {
+        c.collect()
+      }.getMessage
+      // Caused by: java.lang.IllegalArgumentException:
+      // The buffer size must be greater than 0 and less than ${Int.MaxValue}.
+      assert(excMsg.contains("The buffer size must be greater than 0 " +
+        s"and less than ${Int.MaxValue}."))
+      sc0.stop()
+      conf.remove(SHUFFLE_DISK_WRITE_BUFFER_SIZE)
+    }
+  }
```

Comment:
Thanks for adding these test cases! Could you do me a favor and run each test case in a separate branch without any extra change? and post what are the current behaviors?